### PR TITLE
Extend login command to allow specifying Pipeline endpoint and defaul…

### PR DESCRIPTION
**Pull request by stoader**
…t organization as well

### CODE REVIEW

File: internal/cli/command/login.go [hunk 1]
```diff
@@ -79,6 +87,15 @@ func runLogin(banzaiCli cli.Cli, options loginOptions) error {
 		viper.Set("pipeline.basepath", endpoint)
 
 		var orgID int32
+		var orgFound bool
+
+		if options.orgName != "" {
+			orgs := input.GetOrganizations(banzaiCli)
+
+			if orgID, orgFound = orgs[options.orgName]; !orgFound {
+				log.Fatalf("organization with name %q not exists", options.orgName)
```
Comment (orymate) [inline internal/cli/command/login.go]:
```suggestion
				return errors.New(fmt.Sprintf("organization %q doesn't exist", options.orgName))
```

File: internal/cli/command/login.go [hunk 2]
```diff
@@ -79,6 +87,15 @@ func runLogin(banzaiCli cli.Cli, options loginOptions) error {
...
+			}
+		}
 
 		if banzaiCli.Interactive() {
```
Comment (orymate) [inline internal/cli/command/login.go]: else if (don't ask for it if command line option was specified)

File: internal/cli/input/organization.go [hunk 3]
```diff
@@ -55,3 +55,20 @@ func GetOrganization(banzaiCli cli.Cli) int32 {
 
 	return id
 }
+
+// GetOrganizations returns a map with the list of organizations
+// where the key is the organization name and the value is the id
+func GetOrganizations(banzaiCli cli.Cli) map[string]int32 {
```
Comment (orymate) [inline internal/cli/input/organization.go]:
```suggestion
func GetOrganizations(banzaiCli cli.Cli) (map[string]int32, err) {
```

don't exit the program if the orgs can't be listed (this function is not interactive in nature)

File: internal/cli/command/login.go [hunk 4]
```diff
@@ -79,8 +87,18 @@ func runLogin(banzaiCli cli.Cli, options loginOptions) error {
...
+		var orgFound bool
+
+		if options.orgName != "" {
+			orgs, err := input.GetOrganizations(banzaiCli)
+			if err != nil {
+				return emperror.Wrap(err, "could not get organizations")
+			}
+
+			if orgID, orgFound = orgs[options.orgName]; !orgFound {
+				return errors.New(fmt.Sprintf("organization %q doesn't exist", options.orgName))
```
Comment (sagikazarmark) [inline internal/cli/command/login.go]:
```suggestion
				return errors.Errorf("organization %q doesn't exist", options.orgName)
```

File: internal/cli/input/organization.go (hunk 3 above)
Comment (sagikazarmark) [reply internal/cli/input/organization.go]: `err` -> `error`

File: internal/cli/command/login.go [hunk 5]
```diff
@@ -47,12 +50,17 @@ func NewLoginCommand(banzaiCli cli.Cli) *cobra.Command {
 	flags := cmd.Flags()
 
 	flags.StringVarP(&options.token, "token", "t", "", "Pipeline token to save")
+	flags.StringVarP(&options.endpoint, "endpoint", "e", "", "Pipeline API endpoint to save")
+	flags.StringVarP(&options.orgName, "organization", "", "", "name of the default organization for current context")
```
Comment (orymate) [inline internal/cli/command/login.go]:
```suggestion
	flags.StringVarP(&options.orgName, "organization", "", "", "Name of the organization to select as default")
```

**Pull request merged**